### PR TITLE
PrineMenuDONE

### PR DESCRIPTION
**Pull request by ArtemNemtsov**

### CODE REVIEW

File: kitsune/menu/main.cpp [hunk 1]
```diff
@@ -39,19 +39,21 @@ void PrintMenu(const MenuList<string>& menu)
...
-        else cout<<string(60, ' ')<<res[i] << endl; 
+        else cout<<string(40, ' ')<<res[i] << endl; 
    }
 }
 
 int main()
 {
     MenuList<string> menu;
     LoadMenuFromFile(menu, "Menu.txt");
+    menu.ActivateNext();
```
Comment (eaniconer) [inline kitsune/menu/main.cpp]: This was for example It's time to remove it.

File: kitsune/menu/main.cpp [hunk 2]
```diff
@@ -39,19 +39,21 @@ void PrintMenu(const MenuList<string>& menu)
     vector<string> res = menu.GetItems();
     for(int i = 0; i < res.size(); i++)// ShowFirstTime
     {
-        if(i==0 )
+        int index=menu.ActivatedItemIndex();
```
Comment (eaniconer) [inline kitsune/menu/main.cpp]: style:
`Type var = value;`

File: kitsune/menu/main.cpp [hunk 3]
```diff
@@ -39,19 +39,21 @@ void PrintMenu(const MenuList<string>& menu)
...
+        if(i==index)
```
Comment (eaniconer) [inline kitsune/menu/main.cpp]: style:
if (i == index)

File: kitsune/menu/main.cpp [hunk 4]
```diff
@@ -39,19 +39,21 @@ void PrintMenu(const MenuList<string>& menu)
...
         {
             cout <<string( 10, '\n' );
```
Comment (eaniconer) [inline kitsune/menu/main.cpp]: style:
`cout << string(40, ' ');`

File: kitsune/menu/menu.h [hunk 5]
```diff
@@ -26,13 +26,13 @@ namespace kitsune
         /*! Activate next item */
         void ActivateNext() 
         {
-            // your code here
+            activatedIndex_++;
```
Comment (eaniconer) [inline kitsune/menu/menu.h]: What if menu consist of 3 items and the 3rd item is activated. If you do `ActivateNext` index will equals 3, but there is no items with such index. 
Go through indices by cycle: 0 -> 1 -> 2 -> 0 -> 1 -> 2 -> and go on

File: kitsune/menu/menu.h [hunk 6]
```diff
@@ -26,13 +26,13 @@ namespace kitsune
...
         }
 
         /* Activates previous item. */
         void ActivatePrev() 
         {
-            // your code here
+            activatedIndex_--;
```
Comment (eaniconer) [inline kitsune/menu/menu.h]: The same problem as for `ActivateNext`

File: kitsune/menu/main.cpp (hunk 4 above)
Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: Why? Its not a space, its a lines string( 10, '\n' );

Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: And 10 lines its about centre
no?

Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: ok.Done

File: kitsune/menu/main.cpp (hunk 3 above)
Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: done

File: kitsune/menu/main.cpp (hunk 2 above)
Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: Done

File: kitsune/menu/main.cpp (hunk 1 above)
Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: Done

File: kitsune/menu/menu.h (hunk 5 above)
Comment (ArtemNemtsov) [reply kitsune/menu/menu.h]: if (activatedIndex_ != 3 )
            {
                activatedIndex_++;
            }

Comment (ArtemNemtsov) [reply kitsune/menu/menu.h]: Right?

File: kitsune/menu/menu.h (hunk 6 above)
Comment (ArtemNemtsov) [reply kitsune/menu/menu.h]: if (activatedIndex_ != 0 )
            {
                activatedIndex_--;
            } I decide same

File: kitsune/menu/menu.h (hunk 5 above)
Comment (eaniconer) [reply kitsune/menu/menu.h]: You don't know exactly how many items in menu. Number 3 was for example. I would be 4,5,6 and even 0.

Comment (eaniconer) [reply kitsune/menu/menu.h]: In general case you have n items. And the switch should be cycled: 
-1 -> 0 -> 1 -> 2 -> .... -> n -> 0 -> 1 - > 2 -> ... -> n -> 0

File: kitsune/menu/menu.h (hunk 6 above)
Comment (eaniconer) [reply kitsune/menu/menu.h]: Here the switch should be cycled too.
-1 -> n -> (n-1) -> ... -> 2 -> 1 -> 0 -> n -> (n-1) -> ...

File: kitsune/menu/main.cpp [hunk 7]
```diff
@@ -37,23 +37,22 @@ void PrintMenu(const MenuList<string>& menu)
...
         }
-        else cout<<string(40, ' ')<<res[i] << endl; 
+        else cout << string(40, ' ') << res[i] << endl; 
    }
 }
 
 int main()
 {
     MenuList<string> menu;
     LoadMenuFromFile(menu, "Menu.txt");
```
Comment (eaniconer) [inline kitsune/menu/main.cpp]: Rename function:
`LoadMenuFromFile` to `PopulateMenuFromFile`

Comment (ArtemNemtsov) [reply kitsune/menu/main.cpp]: done

**Pull request merged**